packages/date-time/src/components/CalendarDayGrid/CalendarDayGrid.styles.ts

```diff
@@ -104,12 +105,33 @@ export const styles = (props: ICalendarDayGridStyleProps): ICalendarDayGridStyle
       fontWeight: FontWeights.regular,
       color: palette.neutralPrimary,
       cursor: 'pointer',
+      position: 'relative',
       selectors: {
+        [HighContrastSelector]: {
+          color: 'WindowText!important',
+          backgroundColor: 'Window!important',
```

Comment:
we should avoid setting `!important`. we should use minimal specificity to achieve the goal here. this makes very hard for user to override (not sure if they ever need, but still it's not a good practice in general).

Reply:
I removed a few of them. I run into css specificity issues with merge-styles trying to remove the rest. Specifically, I have a classname `dayCell`, which applies a backgroundColor. I have another class `daySelected`, which applies a different background color. Both classes get put on the same <div>.

The color that actually ends up getting applied depends on the order the classNames end up getting written to the generated CSS. If you open the storybook example, and flip from the Calendar to WeeklyDayPicker examples, you'll see that the order gets flipped for the two classes, whereas I would prefer if `daySelected` styles always win. In pure CSS, you can use a specificity hack like defining `.daySelected.daySelected`, but as far as I know you can't do that in mergestyles. So I end up just putting !important on the backgroundColor, and therefore also put !important on the backgroundColor when defining the high contrast settings.

If you know of a way to override the specificity in merge styles defined in our own component, please let me know and I'll try to remove the other !important tags in a future change.